core/space/services/services_keypair.go

```diff
@@ -5,38 +5,28 @@ import (
 	"encoding/hex"
 
 	"github.com/FleekHQ/space-daemon/core/keychain"
-	"github.com/FleekHQ/space-daemon/core/space/domain"
 	"github.com/FleekHQ/space-daemon/core/textile/hub"
 )
 
-func (s *Space) GenerateKeyPair(ctx context.Context, useForce bool) (domain.KeyPair, error) {
+func (s *Space) GenerateKeyPair(ctx context.Context, useForce bool) (string, error) {
+	var mnemonic string
+	var err error
 	if useForce {
-		return s.generateKeyPairWithForce(ctx)
-	}
-
-	kc := keychain.New(s.store)
-	if pub, priv, err := kc.GenerateKeyPair(); err != nil {
-		return domain.KeyPair{}, err
+		mnemonic, err = s.keychain.GenerateKeyFromMnemonic(keychain.WithOverride())
 	} else {
-
-		return domain.KeyPair{
-			PublicKey:  hex.EncodeToString(pub),
-			PrivateKey: hex.EncodeToString(priv),
-		}, nil
+		mnemonic, err = s.keychain.GenerateKeyFromMnemonic()
 	}
+	if err != nil {
+		return "", err
+	}
+
+	return mnemonic, nil
 }
 
-func (s *Space) generateKeyPairWithForce(ctx context.Context) (domain.KeyPair, error) {
-	kc := keychain.New(s.store)
-	if pub, priv, err := kc.GenerateKeyPairWithForce(); err != nil {
-		return domain.KeyPair{}, err
-	} else {
+func (s *Space) RestoreKeyPairFromMnemonic(ctx context.Context, mnemonic string) error {
+	_, err := s.keychain.GenerateKeyFromMnemonic(keychain.WithMnemonic(mnemonic), keychain.WithOverride())
```

Comment:
Since this can override something they already have setting maybe before restoring, the FE should give a warning that restoring will override the key if already generated right?

Reply:
I think it makes sense to show a warning in the UI, but if you are restoring a key you probably don't care about any existing one, so that's why I defaulted to override. Probably going to change a little bit once we implement allowing multiple keys (https://app.clubhouse.io/terminalsystems/story/17248/app-authorization-flow).

From the current FE perspective, if you are showing the mnemonic input that means the daemon is already running without a key pair, otherwise it would show you as logged in.